docs/DesignDocs/AttestationQPLEnforcementSupport.md

```diff
@@ -0,0 +1,70 @@
+Attestation SGX QPL Enforcement Support
+====
+
+This proposal is to make a change to `oe_verify_evidence`, so that it can support policy-based SGX QPL (Quote Provider Library) specific enhancement, such as allowing the OE SDK applications to pass third party cloud specific parameters to the SGX QPL for more advanced quote verification.
+
+Motivation
+----
+
+Intel SGX rolls out PCK certificate and collateral information about every six months.
+There is a scenario where a quote signed by the last version of PCK certificate cannot be verified with the latest version of collateral information.
+This causes unnecessary attestation failures when new versions of PCK certificate and collateral information are rolled out.
+To deal with this issue, the SGX QPL needs to return any version of collateral information based on additional query parameters (referred to as "baselines").
+With that, a new API called `sgx_ql_get_quote_verification_collateral_with_params` was introduced in the SGX QPL.
+Changes are also needed in the OE SDK to call this new API.
+
+Goals:
+ - Applications can call `oe_verify_evidence` and pass third party specific parameters to do customized verification.
+ - Changing the contract for the additional parameters does not need OE SDK changes.
+
+> The contract for the additional parameters are between OE SDK applications and the SGX third party cloud collateral caching services.
+
+User Experience
+----
+
+A new policy type `OE_POLICY_ENDORSEMENTS_BASELINE` is added to current policy types.
+Only one policy in this type can be specified for `oe_verify_evidence` API.
+The data for this policy is an opaque blob for OE SDK.
+
+Specification
+----
+
+### New policy type
+
+- `OE_POLICY_ENDORSEMENTS_BASELINE` in `oe_policy_type_t`
+- The data for this policy is a blob.
+
+### New APIs / OCalls
+
+Though there is no API signature changes for `oe_verify_evidence`, some internal APIs that potentially get used by tools need to be changed.
+For backward compatibility, the following internal APIs will be added to ensure all existing code can still work,
```

Comment:
I don't follow. They're _internal_, so all existing code should be in this repo and can just be changed, no?  Or is the issue that we have not-internal APIs that need to be deprecated?